proposals/0173-Read-Generic-Network-Signal-data.md

```diff
@@ -16,277 +16,306 @@ To enable OEM proprietary mobile apps to read raw CAN/Network data by specifying
 
-    <param name="gps" type="GPSData" mandatory="false">
-      <description>See GPSData</description>
-    </param>
-        
-	<!-- Not showing other params -->
-	
-	<param name="networkData" type="NetworkData" mandatory="false" minsize="1" maxsize="500" maxLength="200" array="true">
-      <description>
-      	Returned Network Data value for a paramId
-      </description>
-    </param> 
+* SDL Proxy needs to be able to download the new JSON schema for vehicle data and add a manager to allow requests for new vehicle data items in a systematic way.
 
-</function>
+* APIs would need a new Request/Response API to act as Generic Request Response for _GetVehicleData/SubscribeVehicleData/UnsubscribeVehicleData_
 
-<function name="OnVehicleData" functionID="OnVehicleDataID" messagetype="notification">
-    <description>Callback for the periodic and non periodic vehicle data read function.</description>
-    <param name="gps" type="GPSData" mandatory="false">
-      <description>See GPSData</description>
-    </param>
-     
-	<!-- Not showing other params -->
-	
-	<param name="networkData" type="NetworkData" mandatory="false" minsize="1" maxsize="500" maxLength="200" array="true">
-      <description>
-      	Returned Network Data value for a paramId
-      </description>
-    </param> 
-	
-  </function>
-  
-  
-<struct name="NetworkData">
-	<param name="paramId" type="Integer" minvalue="1" maxvalue="65535" mandatory="true">
-      <description>Parameter ID for the request.</description>
-    </param>
-	
-	<param name="value" type="String" mandatory="false" maxlength="200" array="false">
-      <description>
-      	Raw Network data HEX string as returned from vehicle.
-      </description>
-    </param>
-</struct>
-```
+## Alternatives considered
+* Instead of using JSON to define vehicle data items schema, we can use either of following:
+  * XML:
+    * This would reuse the XML based design with adding "reference" as another attribute or as a sub-element. Instead of sending a JSON file from ARMR to SYNC we would send an XML file that would be like a subset of the mobile API. 
+    * But this would need changes at PT server side as Policy server is configured to serve JSON as part of PT update.
+  * Precompiled library plug-ins:
+    * Another idea is to automatically generate code and compile this code into a dynamic link library. Instead of sending a JSON or XML file this library would be transferred to SYNC as part of policy update or OTA. Core would load this library and execute this library for vehicle data.
```

Comment:
Change to:
Another idea is to automatically generate code and compile this code into a dynamic link library. Instead of sending a JSON or XML **file, this** library would be transferred to SYNC as part of policy update or OTA. Core would load this library and execute this library for vehicle data.